config/routes.rb

```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  get '/', to: 'users#index', as: :users
+  get '/users/:id', to: 'users#show', as: :user
+  get '/users/:id/posts', to: 'posts#index', as: :user_posts
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: :user_post
 end
```

Comment:
- [Optional ] There's a concise and powerful way to define your routes using a more fine-grained approach, which can make your code cleaner and more explicit. Here is an example.

```

  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end

```